File: engineering/codebase/smartqr.frontend-services/src/domain/codes/rules/enums/RuleConditionType.ts
/** Defines the signal a routing rule matches on (mirrors backend `conditionType`). */
export const RuleConditionType = {
  /** Refers to the catch-all — always matches, so it carries the code's destination when no other rule does. */
  Default: "default",
  /** Refers to the scanning device kind (iOS · Android · Desktop). */
  Device: "device",
  /** Refers to the visitor's country. */
  Country: "country",
  /** Refers to the visitor's preferred language. */
  Language: "language",
  /** Refers to the local time of day at scan. */
  TimeOfDay: "timeOfDay",
} as const;

export type RuleConditionType = (typeof RuleConditionType)[keyof typeof RuleConditionType];

File: engineering/codebase/smartqr.backend-services/SmartQr.Domain/Codes/Core/Enums/RuleConditionType.cs
namespace SmartQr.Domain.Codes.Core.Enums;

/// <summary>Defines the signal a routing rule matches on.</summary>
public enum RuleConditionType
{
    /// <summary>Represents a match on the device class (e.g. <c>Ios</c>).</summary>
    Device,

    /// <summary>Represents a match on the ISO country code from IP geo (e.g. <c>US</c>).</summary>
    Country,

    /// <summary>Represents a match on the primary language tag from Accept-Language (e.g. <c>ru</c>).</summary>
    Language,

    /// <summary>Represents a match on a daily time window <c>HH:mm-HH:mm</c>.</summary>
    TimeOfDay,
}
